driver/src/main/java/org/neo4j/driver/v1/GraphDatabase.java

```diff
@@ -131,4 +133,44 @@ public static Driver driver( URI uri, AuthToken authToken, Config config )
 
         return new DriverFactory().newInstance( uri, authToken, routingSettings, retrySettings, config );
     }
+
+    /**
+     * Try to create a bolt+routing driver from the first available address.
+     * This is wrapper for the {@link #driver} method that finds the first
+     * server to respond positively.
+     *
+     * @param addresses a list of server addresses for Neo4j instances
+     * @param authToken authentication to use, see {@link AuthTokens}
+     * @param config user defined configuration
+     * @return a new driver instance
+     */
+    public static Driver routingDriverFromFirstAvailableAddress( List<String> addresses, AuthToken authToken, Config config )
+    {
+        for( String address: addresses )
+        {
+            try
+            {
+                return driver( "bolt+routing://" + address, authToken, config );
+            }
+            catch( ServiceUnavailableException e )
+            {
+                // try the next one
+            }
+        }
+        throw new ServiceUnavailableException( "Failed to discover an available server" );
+    }
+
+    /**
+     * Try to create a bolt+routing driver from the first available address.
+     * This is wrapper for the {@link #driver} method that finds the first
+     * server to respond positively.
+     *
+     * @param addresses a list of server addresses for Neo4j instances
+     * @param authToken authentication to use, see {@link AuthTokens}
+     * @return a new driver instance
+     */
+    public static Driver routingDriverFromFirstAvailableAddress( List<String> addresses, AuthToken authToken )
```

Comment:
Maybe it is ok to not add this method because we do not add another overload with just list of addresses?